File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/dataimportvb/vb/importwithcreate3.vb
' <snippetimportwithcreate3>


 ' Retrieve the first two distinct values for column 1 from the parse table.
 ' NOTE: You must create the parse table first using the ParseImport message.
 ' The parse table is not accessible after ImportRecordsImportResponse is called.
 Dim distinctValuesRequest As New GetDistinctValuesImportFileRequest() With
     {
         .columnNumber = 1,
         .ImportFileId = importFileId,
         .pageNumber = 1,
         .recordsPerPage = 2
     }
 Dim distinctValuesResponse As GetDistinctValuesImportFileResponse =
     CType(_serviceProxy.Execute(distinctValuesRequest), 
         GetDistinctValuesImportFileResponse)

 ' Output the distinct values.  In this case: (column 1, row 1) and (column 1, row 2).
 Dim cellNum As Integer = 1
 For Each cellValue As String In distinctValuesResponse.Values
     Console.WriteLine("(1, " &amp; cellNum.ToString() &amp; "): " &amp; cellValue)
     Console.WriteLine(cellValue)
     cellNum += 1
 Next cellValue

' </snippetimportwithcreate3>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/dataimport/cs/importwithcreate3.cs
// <snippetimportwithcreate3>


 // Retrieve the first two distinct values for column 1 from the parse table.
 // NOTE: You must create the parse table first using the ParseImport message.
 // The parse table is not accessible after ImportRecordsImportResponse is called.
 GetDistinctValuesImportFileRequest distinctValuesRequest = new GetDistinctValuesImportFileRequest()
 {
     columnNumber = 1,
     ImportFileId = importFileId,
     pageNumber = 1,
     recordsPerPage = 2,
 };
 GetDistinctValuesImportFileResponse distinctValuesResponse =
     (GetDistinctValuesImportFileResponse)_serviceProxy.Execute(distinctValuesRequest);

 // Output the distinct values.  In this case: (column 1, row 1) and (column 1, row 2).
 int cellNum = 1;
 foreach (string cellValue in distinctValuesResponse.Values)
 {
     Console.WriteLine("(1, " + cellNum.ToString() + "): " + cellValue);
     Console.WriteLine(cellValue);
     cellNum++;
 }

// </snippetimportwithcreate3>